typescript/cli/src/deploy/warp.ts

```diff
@@ -0,0 +1,342 @@
+    'interchainGasPaymaster',
+  ];
+  let hasShownInfo = false;
+  for (const [chain, token] of Object.entries(configMap)) {
+    for (const field of requiredRouterFields) {
+      if (token[field]) continue;
+      if (!hasShownInfo) {
+        logBlue(
+          'Some router fields are missing. Please enter them now, add them to your warp config, or use the --core flag to use deployment artifacts.',
+        );
+        hasShownInfo = true;
+      }
+      const value = await input({
+        message: `Enter ${field} for ${getTokenName(token)} token on ${chain}`,
+      });
+      if (!value) throw new Error(`Field ${field} required`);
+      token[field] = value.trim();
+    }
+  }
+
+  log('Token configs ready');
+  return {
+    configMap,
+    metadata: baseMetadata,
+    local: baseChainName,
+    remotes: synthetics.map(({ chainName }) => chainName),
+    isNft: !!isNft,
+  };
+}
+
+interface DeployParams {
+  configMap: ChainMap<TokenConfig & RouterConfig>;
+  isNft: boolean;
+  metadata: MinimalTokenMetadata;
+  local: ChainName;
+  remotes: ChainName[];
+  signer: ethers.Signer;
+  multiProvider: MultiProvider;
+  outPath: string;
+}
+
+async function runDeployPlanStep({
+  configMap,
+  isNft,
+  local,
+  remotes,
+  signer,
+}: DeployParams) {
+  const address = await signer.getAddress();
+  const baseToken = configMap[local];
+  const baseName = getTokenName(baseToken);
+  logBlue('\nDeployment plan:');
+  logGray('===============:');
+  log(`Transaction signer and owner of new contracts will be ${address}`);
+  log(`Deploying a warp route with a base of ${baseName} token on ${local}`);
+  log(`Connecting it to new synthetic tokens on ${remotes.join(', ')}`);
+  log(`Using token standard ${isNft ? 'ERC721' : 'ERC20'}`);
+
+  const isConfirmed = await confirm({
+    message: 'Is this deployment plan correct?',
+  });
+  if (!isConfirmed) throw new Error('Deployment cancelled');
+}
+
+async function executeDeploy(params: DeployParams) {
+  logBlue('All systems ready, captain! Beginning deployment...');
+
+  const { configMap, isNft, multiProvider, outPath } = params;
+
+  const [contractsFilePath, tokenConfigPath] = prepNewArtifactsFiles(outPath, [
+    { filename: 'warp-deployment', description: 'Contract addresses' },
+    { filename: 'warp-ui-token-config', description: 'Warp UI token config' },
+  ]);
+
+  const deployer = isNft
+    ? new HypERC721Deployer(multiProvider)
+    : new HypERC20Deployer(multiProvider);
+
+  const deployedContracts = await deployer.deploy(configMap);
+  logGreen('Hyp token deployments complete');
+
+  log('Writing deployment artifacts');
+  writeTokenDeploymentArtifacts(contractsFilePath, deployedContracts, params);
+  writeWarpUiTokenConfig(tokenConfigPath, deployedContracts, params);
+
+  logBlue('Deployment is complete!');
+  logBlue(`Contract address artifacts are in ${contractsFilePath}`);
+  logBlue(`Warp UI token config is in ${tokenConfigPath}`);
+}
+
+async function fetchBaseTokenMetadata(
```

Comment:
we should merge all this with the token package

Reply:
Agreed, happy to do that next. I wanted to limit this PR to mostly migrating things in from hyp-deploy. I can make those changes now if @nambrot and @avious00 greenlight more work here

Reply:
Punting, left a TODO